src/main/java/org/apache/sysml/runtime/matrix/data/LibMatrixCUDA.java

```diff
@@ -948,4 +983,173 @@ public static boolean isInSparseFormat(MatrixObject mo) {
 			return mo.getGPUObject().isInSparseFormat();
 		return MatrixBlock.evalSparseFormatInMemory(mo.getNumRows(), mo.getNumColumns(), mo.getNnz());
 	}
+	
+	/**
+	 * Performs elementwise multiplication of input vector in and constant using cublasDaxpy
+	 * 
+	 * @param ec
+	 * @param in
+	 * @param constant
+	 * @param outputName
+	 * @param op
+	 * @throws DMLRuntimeException
+	 */
+	public static void vectorScalarMult(ExecutionContext ec, MatrixObject in, double constant, String outputName, GPUEnabledElementwiseOp op) throws DMLRuntimeException {
+		if(isInSparseFormat(in)) {
+			// TODO: Implement sparse vector scalar multiply kernel
+			// throw new DMLRuntimeException("Sparse vector scalar multiply is not supported");
+			((JCudaObject)in.getGPUObject()).sparseToDense();
+		}
+		
+		MatrixObject out = ec.getMatrixObject(outputName);
+	    ec.getDenseMatrixOutputForGPUInstruction(outputName);	// Allocated the dense output matrix
+	    
+		Pointer alpha = pointerTo(constant);
+		if(op == GPUEnabledElementwiseOp.DIVIDE)
+			alpha = pointerTo(1/constant);
+		
+		int incx = 1, incy = 0;
+		int n = (int) in.getNumRows() == 1 ? (int) in.getNumColumns() : (int) in.getNumRows();
+		
+		Pointer A = ((JCudaObject)in.getGPUObject()).jcudaDenseMatrixPtr;
+	    Pointer C = ((JCudaObject)out.getGPUObject()).jcudaDenseMatrixPtr;
+	    
+		JCublas2.cublasDaxpy(cublasHandle, n, alpha, A, incx, C, incy);
+	}
+	
+	/**
+	 * Performs elementwise multiplication or division of input matrix in and constant using cublasDgeam
+	 * 
+	 * @param ec
+	 * @param in
+	 * @param constant
+	 * @param outputName
+	 * @param op
+	 * @param isTransposed
+	 * @throws DMLRuntimeException
+	 */
+	public static void matScalarElementwiseMultDiv(ExecutionContext ec, MatrixObject in, 
```

Comment:
Look at the "control parameter" argument left at vectorScalarMult.
Also, names of the arguments should match [cublasDgeam](http://docs.nvidia.com/cuda/cublas/#cublas-lt-t-gt-geam) since this is a wrapper over it. (**alpha** for constant, **A** for the matrix)